### PR TITLE
Merge CMA into contentful.objc

### PR DESCRIPTION
**Pull request by loudmouth**
This PR migrates [contentful-management.objc](https://github.com/contentful/contentful-management.objc) code base into this repo. This needed to be done as the management SDK was using private headers, for instance `CDAEntry+Private.h` which lived in [contentful.objc](https://github.com/contentful/contentful.objc). Since the prior version of the project used cocoapods `v0.39` which can no longer update the specs repo locally or push pods.

In addition, the testing infrastructure has been remade to handle the migration (which already happened in the delivery SDK) from `NSURLConnection` (deprecated with release of iOS 8) to `NSURLSession`.

### CODE REVIEW

File: ManagementAPI/Private/CMAAsset.m [hunk 1]
```diff
@@ -0,0 +1,154 @@
+//
+//  CMAAsset.m
+//  Pods
+//
+//  Created by Boris Bügling on 28/07/14.
```
Comment (Roblinde) [inline ManagementAPI/Private/CMAAsset.m]: Boris resurrected?

File: docs/undocumented.json [hunk 2]
```diff
@@ -1,19 +1,6 @@
...
-    {
-      "file": "/Users/jpwright/Contentful/objc/SDK/contentful.objc/ContentfulDeliveryAPI/Resources/CDAError.h",
-      "line": 14,
-      "symbol": "CDAErrorDomain",
-      "symbol_kind": "sourcekitten.source.lang.objc.decl.constant",
-      "warning": "undocumented"
-    }
+
   ],
   "source_directory": "/Users/jpwright/Contentful/objc/SDK/contentful.objc"
```
Comment (Roblinde) [inline docs/undocumented.json]: no clue what this is, but should it poin to your local user directory?

File: ManagementTests/SpacesSpec.m [hunk 3]
```diff
@@ -188,53 +188,11 @@
     });
     VCRTestEnd
 
-    VCRTest_it(@"can_create_new_Space_within_specific_Organization")
```
Comment (mariobodemann) [inline ManagementTests/SpacesSpec.m]: Do you have a different test about creating a new space somewhere? Afaik I did not find a test for this api functionality.

File: ContentfulManagementAPI.podspec [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/ruby
+
+Pod::Spec.new do |spec|
+
+  spec.name             = "ContentfulManagementAPI"
+  spec.version          = "0.10.0"
+  spec.summary          = "Objective-C SDK for Contentful's Content Management API."
+  spec.homepage         = "https://github.com/contentful/contentful-management.objc"
+  spec.author           = { "Boris Bügling" => "boris@buegling.com" }
```
Comment (mariobodemann) [inline ContentfulManagementAPI.podspec]: Is this still true? I thought this JP guy does it now...

File: Podfile.lock [hunk 5]
```diff
@@ -34,9 +34,9 @@ PODS:
   - Keys (1.0.0)
   - OCMock (3.4)
   - PDKTCollectionViewWaterfallLayout (0.1)
-  - Realm (2.5.0):
-    - Realm/Headers (= 2.5.0)
-  - Realm/Headers (2.5.0)
+  - Realm (2.5.1):
+    - Realm/Headers (= 2.5.1)
+  - Realm/Headers (2.5.1)
```
Comment (mariobodemann) [inline Podfile.lock]: Confusion is hight: Do you need the headers twice?

File: Gemfile [hunk 6]
```diff
@@ -1,11 +1,11 @@
 source 'http://rubygems.org'
 
-gem 'cocoapods', '= 1.2.0'
+gem 'cocoapods', '~> 1.2.0'
 gem 'xcpretty'
 gem 'xcode-install'
 gem 'cocoapods-keys', '~> 2.0.0'
 gem 'cocoapods-coverage'
-gem 'nokogiri'
+gem 'nokogiri' #, '= 1.6.3.1'
```
Comment (mariobodemann) [inline Gemfile]: Can you remove the comment # if you don't need the version number?

File: .travis.yml [hunk 7]
```diff
@@ -17,6 +17,7 @@ install:
 - bundle exec pod install
 script:
 - "./scripts/travis-build-test.sh"
+- "./scripts/travis-lint.sh"
```
Comment (mariobodemann) [inline .travis.yml]: Do we need the quotes? The look strange to me …

File: ManagementAPI/Private/CMAAsset.m (hunk 1 above)
Comment (loudmouth) [inline ManagementAPI/Private/CMAAsset.m]: Yeah, he created the files—tbh, I don't even like these autogenerated comments at the top as it's just redundant, and useless info: the name of the file which you can see in the file system already, and the creator, who might not have even written the bulk of the contained code.

File: docs/undocumented.json (hunk 2 above)
Comment (loudmouth) [inline docs/undocumented.json]: this is all autogenerated by the the cli i use to generate docs.

File: ContentfulDeliveryAPI/CDAConfiguration.h [hunk 8]
```diff
@@ -16,7 +16,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-extern const NSString* const CDA_DEFAULT_SERVER;
+extern NSString* const CDA_DEFAULT_SERVER;
```
Comment (loudmouth) [inline ContentfulDeliveryAPI/CDAConfiguration.h]: This was a compile error  before and it had to be changed. I'm not sure I understand your questions though. AFAIK, it isnt' writable.

Comment (loudmouth) [inline ContentfulDeliveryAPI/CDAConfiguration.h]: Yes, @mariobodemann, to confirm, it is NOT writable: in this case, the const is actually applied to the value. See this [SO answer](http://stackoverflow.com/a/3196554/4068264)

File: ContentfulDeliveryAPI/CDARequestOperationManager.h [hunk 9]
```diff
@@ -46,4 +46,8 @@
                  success:(CDAObjectFetchedBlock)success
                  failure:(CDARequestFailureBlock)failure;
 
++ (NSData *)sendSynchronousRequest:(NSURLRequest *)request
```
Comment (loudmouth) [inline ContentfulDeliveryAPI/CDARequestOperationManager.h]: Git is weird, it's not new.

File: ContentfulDeliveryAPI/CDAConfiguration.h (hunk 8 above)
Comment (loudmouth) [inline ContentfulDeliveryAPI/CDAConfiguration.h]: Reverting...

File: Podfile.lock (hunk 5 above)
Comment (loudmouth) [inline Podfile.lock]: Realm docs say to add `Realm/Headers` for test targest.

File: ContentfulManagementAPI.podspec (hunk 4 above)
Comment (loudmouth) [inline ContentfulManagementAPI.podspec]: Adding my name :-)

File: ContentfulDeliveryAPI/Resources/CDAResource.m [hunk 10]
```diff
@@ -174,7 +174,7 @@ -(NSString *)identifier {
     return identifier;
 }
 
--(instancetype)initWithDictionary:(NSDictionary *)dictionary
+-(instancetype)initWithDictionary:(nonnull NSDictionary *)dictionary
```
Comment (loudmouth) [inline ContentfulDeliveryAPI/Resources/CDAResource.m]: Reverting...

**Pull request merged**